presto-hive/src/main/java/com/facebook/presto/hive/util/SerDeUtils.java

```diff
@@ -289,10 +291,16 @@ private static long formatDateAsLong(Object object, DateObjectInspector inspecto
         return TimeUnit.MILLISECONDS.toDays(millisUtc);
     }
 
-    private static long formatTimestampAsLong(Object object, TimestampObjectInspector inspector)
+    private static long formatTimestampAsLong(Object object, TimestampObjectInspector inspector, DateTimeZone hiveStorageTimeZone)
     {
         Timestamp timestamp = getTimestamp(object, inspector);
-        return timestamp.getTime();
+        long parsedJvmMillis = timestamp.getTime();
```

Comment:
This is the part with the change. Logic is copied from https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java#L320-L333